projects/jdwp/serializers/serializer_codegen.py

```diff
@@ -0,0 +1,99 @@
+import re
+import os
+from projects.jdwp.defs.schema import Type, Struct, CommandSet, Command, Field
+from projects.jdwp.defs.command_sets.reference_type import ReferenceType
+
+
+class CommandSerializerGenerator:
+    def __init__(self):
+        self.serializer_code: str = ""
+
+    def generate_field_serializer(self, field: Field) -> str:
+        serializer_code: str = ""
+        if isinstance(field, Struct):
+            for subfield in field.fields:
+                serializer_code += self.generate_field_serializer(subfield)
+        elif field.type == Type.INT:
+            serializer_code = (
+                f"        serialized_data += command.{field.name}.to_bytes(4, 'big')"
+            )
+        elif field.type == Type.STRING:
+            serializer_code = (
+                f"        serialized_data += command.{field.name}.encode('utf-8')"
+            )
+        elif field.type == Type.OBJECT_ID:
+            serializer_code = (
+                f"        serialized_data += command.{field.name}.to_bytes(8, 'big')"
```

Comment:
This is not that simple. The spec says:
> Object ids, reference type ids, field ids, method ids, and frame ids may be sized differently in different target VM implementations. Typically, their sizes correspond to size of the native identifiers used for these items in JNI and JVMDI calls. The maximum size of any of these types is 8 bytes. The "idSizes" command in the VirtualMachine command set is used by the debugger to determine the size of each of these types.

This is one more reason to have input/output stream abstractions. If we do that then we'll be able to configure them with output of "idSizes" command.